api/candlepin-api-spec.yaml

```diff
@@ -1124,6 +1124,390 @@ paths:
+      security: []
+      responses:
+        200:
+          description: Returns a role
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoleDTO'
+        400:
+          description: Role name is null or empty
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role name is null or empty
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        404:
+          description: Role not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role not found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    put:
+      description: Updates Role. To avoid race conditions, we do not support updating the user or permission
+        collections. Currently this call will only update the role name. See the specific nested POST/DELETE
+        calls for modifying users and permissions.
+      tags:
+        - Roles
+      operationId: updateRole
+      security: []
+      parameters:
+        - name: role_name
+          in: path
+          description: Role name
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Role
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RoleDTO'
+      responses:
+        200:
+          description: Returns updated role
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoleDTO'
+        400:
+          description: Role name is null or empty
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role name is null or empty
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        404:
+          description: Role not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role not found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    delete:
+      description: Removes a Role
+      tags:
+        - Roles
+      operationId: deleteRoleByName
+      parameters:
+        - in: path
+          name: role_name
+          description: Role name
+          required: true
+          schema:
+            type: string
+      security: []
+      responses:
+        200:
```

Comment:
We should return 204 as we are not returning any data.

Reply:
Corrected.